src/pubsub.c

```diff
@@ -30,8 +30,98 @@
 #include "server.h"
 #include "cluster.h"
 
+/* Structure to hold the pubsub related metadata. Currently used
+ * for pubsub and pubsublocal feature. */
+typedef struct pubsubtype {
+    int local;
+    dict *(*serverPubSubChannels)();
+    dict *(*clientPubSubChannels)(client*);
+    int (*subscriptionCount)(client*);
+    robj *(*subscribeMsg)();
+    robj *(*unsubscribeMsg)();
```

Comment:
Still not clear why this can't just be a pointer to an robj, and instead has to be a function that returns an robj.

Reply:
If I were to use the variables from `shared`, then there is compile time error. If we want to use an robj, I can create a new object for the struct `pubsubtype`.

Reply:
I misunderstood your comment, updated the code. Thanks for the suggestion.